### PR TITLE
Fix PY-1177 Control flow analysis doesn't detect while/break

### PR DESCRIPTION
**Pull request by fitermay**

### CODE REVIEW

**Comment by vlasovskikh** [comment]
Thanks for your PR! I'll review it later.

We would like to unify our static evaluation code (`PyConstantExpressionEvaluator`, `PyEvaluator`, and others) and make it explicit what get evaluated and how much it affects performance.

**Comment by fitermay** [comment]
Does it mean I need to change something or is it just a general comment?

**Comment by vlasovskikh** [comment]
@fitermay No, your code looks OK at first glance.

**Comment by fitermay** [comment]
ping?

File: python/src/com/jetbrains/python/codeInsight/controlflow/PyControlFlowBuilder.java [hunk 1]
```diff
@@ -738,7 +738,7 @@ public void visitPyAssertStatement(final PyAssertStatement node) {
     super.visitPyAssertStatement(node);
     final PyExpression[] args = node.getArguments();
     // assert False
-    if (args.length >= 1 && PyConstantExpressionEvaluator.evaluate(args[0]) == Boolean.FALSE) {
+    if (args.length >= 1 && PyConstantExpressionEvaluator.evaluateBoolean(args[0], true)) {
```
Comment (sproshev) [inline python/src/com/jetbrains/python/codeInsight/controlflow/PyControlFlowBuilder.java]: I suppose you forgot about "!" before PyConstantExpressionEvaluator.evaluateBoolean(args[0], true) :)

**Pull request closed without merging**